Update Sentry CLI for MSBuild (#6183)
getsentry/sentry-docs: src/platforms/dotnet/common/configuration/msbuild.mdx

---
title: MSBuild Setup
description: "Configure MSBuild properties in your .NET project to automatically use the Sentry CLI."
sidebar_order: 10
---

> The [Sentry CLI](/product/cli) provides many useful features that can improve your experience with Sentry.
> For .NET, we can use it to upload [debug information files](/product/cli/dif) to Sentry, such as .NET `PDB` symbol files.
> This allows Sentry to show a more complete stack trace, including filenames and line numbers.
> You can also use it to upload *source code*, which enables Sentry to display [source context](../../data-management/debug-files/source-context/)
> when viewing stack traces.

Reviewer: ```suggestion
The [Sentry CLI](/product/cli) has many features that can improve your Sentry experience.
For .NET, you can use it to upload [debug information files](/product/cli/dif), such as .NET `PDB` symbol files.
This allows Sentry to show a more complete stack trace, including file names and line numbers.
You can also use it to upload *source code*, which enables Sentry to display [source context](../../data-management/debug-files/source-context/)
when viewing stack traces.
```

> While you can always run `sentry-cli` from the command line manually, we have made it easier by including it in the `Sentry`
> NuGet package, and providing properties to enable it from your build.

Reviewer: ```suggestion
While you can always run `sentry-cli` from the command line manually, we've made it easier by including it in the `Sentry`
NuGet package, and providing properties to enable it from your build.
```

### Additonal MSBuild Properties

> There are a few extra properties available to control the usage of the Sentry CLI:

Reviewer: ```suggestion
There are a few additional properties available to control the Sentry CLI:
```

</ConfigKey>

<ConfigKey name="SentryUploadSources">

> Controls uploading source code files to Sentry during the build.  Defaults to `false`.
> Set `true` to opt-in to enable [source context](../../data-management/debug-files/source-context/) when viewing stack traces.

Reviewer: ```suggestion
Controls how source code files are uploaded  to Sentry during the build.  Defaults to `false`.
Set `true` to opt-in to enable [source context](../../data-management/debug-files/source-context/) when viewing stack traces.
```
